TCMSApp/src/client/app/apis/trello/create-project.service.js

```diff
@@ -0,0 +1,55 @@
+(function () {
+    'use strict';
+
+    angular
+        .module('app.trello')
+        .factory('createProjectFactory', createProjectFactory);
+
+    createProjectFactory.$inject = ['Trello', 'logger', 'addProjectFactory', 'fillSidebarFactory'];
+    /* @ngInject */
+    function createProjectFactory(Trello, logger, addProjectFactory, fillSidebarFactory) {
+
+        return {
+            createProjAndOrg: createProjAndOrg
+        };
+
+        function createProjAndOrg (Trello, projectName, projectDescription, vm, modalWindow) {
+
+            Trello.get('members/me/organizations').then(
+                function (res) {
+                    var same = false,
+                        l = res.length,
+                        i;
+                    for (i = 0; i < l; i++) {
+                        if (res[i].displayName === projectName) {
+                            same = true;
+                            break;
+                        }
+                    }
+                    if (same) logger.error('Project with the same name already exists.', '', 'Error');
+                    else {
+                        Trello.post('/organizations', {
+                            name: projectName,
+                            displayName: projectName,
+                            desc: projectDescription
+                        }).then(function (res) {
+                                logger.success('Project ' + projectName + ' created. Description: ' + projectDescription, '', 'Project created');
+                                vm.changeModal(modalWindow);
```

Comment:
you are pushing modal tooooo deep. this logic must be in controller (and then you should not pass here vm and modal)